File: p12.py
def intToRoman(num:int):
    i=0
    res=""
    vals=[["M",1000],["CM",900],["D",500],["CD",400],["C",100],["XC",90],["L",50],["XL",40],["X",10],["IX",9],["V",5],["IV",4],["I",1]]
    while num>0:
        if num<vals[i][1]:
            i+=1
        else:
            num-=vals[i][1]
            res+=vals[i][0]
    return res
print(intToRoman(109))

        
        
        
        

File: p12.c
char *intToRoman(int num)
{
    char *result = (char *)malloc(16);
    int i = 0;
    int values[] = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
    char *symbols[] = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};

    for (int j = 0; j < sizeof(values) / sizeof(values[0]); j++)
    {
        while (num >= values[j])
        {
            num -= values[j];
            i += sprintf(result + i, "%s", symbols[j]);
        }
    }
    return result;
}